ui/imports/shared/status/StatusChatImageLoader.qml

```diff
@@ -6,30 +6,68 @@ import shared.panels 1.0
 import utils 1.0
 
 Item {
+    id: root
+
     property int verticalPadding: 0
     property int imageWidth: 350
     property bool isCurrentUser: false
     property url source
     property bool isActiveChannel: false
     property bool playing: Global.applicationWindow.active && isChatActive
     property bool isAnimated: !!source && source.toString().endsWith('.gif')
-    signal clicked(var image, var mouse)
     property var container
     property alias imageAlias: imageMessage
     property bool allCornersRounded: false
+    property bool isOnline: true // TODO: mark as required when migrating to 5.15 or above
+
+    signal clicked(var image, var mouse)
 
-    id: imageContainer
     width: loadingImageLoader.active ? loadingImageLoader.width : imageMessage.width
     height: loadingImageLoader.active ? loadingImageLoader.height : imageMessage.paintedHeight
 
+    onIsOnlineChanged: {
+        if (!isOnline)
+            return
+
+        if (imageMessage.status === Image.Error) {
+            imageMessage.reloadImage()
+        }
+    }
+
+    Timer {
```

Comment:
I somehow miss the point of this Timer; shouldn't we just react on `isOnline` changes?

Reply:
Is tricky because it takes time from actual going offline to the change `isOnline=false`. So it may happen that we have some missing images because we are offline, but we go online back before the app notices that we were offline. So `isOffline` is permanently true in this case. This timer is added to address the described situation.

Reply:
OK... I see